Automattic/wp-calypso
# Patterns: Fix patterns that could be copied when searching

This pull request seeks to address https://github.com/Automattic/dotcom-forge/issues/6323 by enforcing a limit of three patterns per category that can be copied while being logged out. This fixes the case where it would be possible to copy a locked pattern by looking it up:

![image](https://github.com/Automattic/wp-calypso/assets/594356/c0e096d3-c8cc-4312-b945-29cd70fffb91)

#### Testing instructions

1. Run `git checkout lock/patterns` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/89023#issuecomment-2025796292)
2. Make sure you are logged out from WordPress.com
3. Open the [`Patterns` page](http://calypso.localhost:3000/patterns?grid=1)
4. Search for `contact`
5. Assert that the pattern that can be copied are shown at the top
6. Open the [`Footers Patterns` page](http://calypso.localhost:3000/patterns/footer?grid=1)
7. Assert that only the first three patterns can be copied
8. Search for `Footer: Simple, contact info left` now
9. Assert that the pattern displayed can be copied
10. Search for `Four columns of menus with social`
11. Assert that the pattern displayed cannot be copied
12. Navigate to the [`About Page Patterns` page](http://calypso.localhost:3000/patterns/layouts/about?grid=1)
13. Assert that only the first three patterns can be copied
